credential-viewer-component.js

```diff
@@ -145,6 +146,11 @@ function Ctrl(
         })
         .then(function() {
           $scope.$apply();
+        })
+        .then(function() {
+          if(self.credential.deleted && self.onDelete) {
```

Comment:
Does `self.onDelete` always exist even if it's optional? I can't remember if angular sets it to `angular.noop()` if you don't provide an expression to execute via the HTML.